pkg/proctree/types.go

```diff
@@ -0,0 +1,50 @@
+package proctree
+
+// ProcessInfo is the user facing representation of a process data at a specific time.
+type ProcessInfo struct {
+	Id              int
+	NsId            int
+	Ppid            int
+	NsPpid          int // TODO: is this redundant?
+	UserId          int
+	ContainerId     string
+	Cmd             []string
+	ExecutionBinary FileInfo
+	StartTime       int // First thread fork time. TODO: Can we use the mian thread start time instead?
+	ExecTime        int // Last execve call time
+	ExitTime        int
+	ExistingThreads []int
+	ChildrenIds     []int
+	IsAlive         bool
+}
+
+// ThreadInfo is the user facing representation of a thread data at a specific time.
+type ThreadInfo struct {
+	HostId        int
```

Comment:
No need for Host if you call the below field NsId. Making it Id will also be more consistent with Id in the ProcessInfo